name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/HTMLTableQuoteFeed.java

```diff
@@ -460,7 +461,8 @@ protected String getUserAgent()
 
         // if no quotes could be extract, log HTML for further analysis
         if (prices.isEmpty())
-            errors.add(new IOException(MessageFormat.format(Messages.MsgNoQuotesFoundInHTML, url, document.html())));
+            errors.add(new IOException(MessageFormat.format(Messages.MsgNoQuotesFoundInHTML, url,
+                            Jsoup.clean(document.html(), Whitelist.relaxed()))));
```

Comment:
> Im Fehlerfall wird das Dokument um Formatierungen bereinigt.

Die Änderung ist unproblematisch, weil das nur die Anzeige des HTML im Fehlerfall betrifft.

Ich frage mich, was der Nutzen ist. Klar, das HTML ist einfacher zu lesen. Auf der anderen Seite sieht man nicht mehr genau was der HTMLTableQuoteFeed verarbeitet hat. Kann es passieren, dass das weitere Tags in der Zelle stehen die verhindern das PP den Wert parsen kann, die aber dann per "clean" rausgenommen werden?

Andrerseits bekommen wir die Fehlermeldungen ja meist mit URL.

Reply:
🤔 Das halte ich an dieser Stelle für unkritisch da das Ziel ja primär die HTML Tabelle ist. Da wäre allenfalls der ROWSPAN / COLSPAN Tag flöten geht, aber darauf reagiert der Konverter schon jetzt nicht. Die Whitelist lässt sich im Bedarfsfall beliebig erweitern, solange die reinen Table-Tags gebraucht werden ist mE der optische Schnickschnack derzeit nur Makulatur.

 a, b, blockquote, br, caption, cite, code, col, colgroup, dd, div, dl, dt, em, h1, h2, h3, h4, h5, h6, i, img, li, ol, p, pre, q, small, span, strike, strong, sub, sup, table, tbody, td, tfoot, th, thead, tr, u, ul